homeassistant/bootstrap.py

```diff
@@ -182,6 +127,74 @@ def _setup_component(hass: core.HomeAssistant, domain: str, config) -> bool:
         return True
 
 
+def prepare_setup_component(hass: core.HomeAssistant, config: dict,
```

Comment:
This is slightly different from what I envisioned in #2912  - to have a function for config validation only

But I'm taking my sweet time, so let me rebase that PR from this one and continue working on it